File: 01/p01.nim
import strutils, sequtils, math, tables

let
  strNums = readFile("input.txt").strip().splitLines()  
  nums = strNums.map(parseInt)       

let
  sumNums = sum(nums)
  
echo sumNums


var freqs = initTable[int, bool]()
var freq = 0
var found = false
while true:
  for f in nums:
    freq += f
    if freqs.hasKey(freq):
      echo freq
      found = true
      break
    freqs[freq] = true
  if found:
    break

File: 01/p01.kt
import java.io.File

val p01 =  fun(input: List<String>) {
    var numbers = input.map {it.toInt()}

    val sum = numbers.reduce{sum, it->sum+it}
    println(sum)

    var freq = 0
    var freqs = HashSet<Int>(freq)
    for (number in numbers.asSequence().infinite()){
        freq += number
        if (!freqs.add(freq)){
            println(freq)
            return
        }
    }
}